client/components/user/UserDataCard.tsx

```diff
@@ -18,60 +19,35 @@ const UserDataCard = ({
   favorited_by,
   favorites_count,
   createdAt,
-  state,
-  setState,
   index,
   differentUser,
 }) => {
-  const dispatch = useDispatch();
-
-  /**
-   * Updates the state at the specified index with the given boolean value.
-   *
-   * @param {number} index - The index of the state to be updated.
-   * @param {boolean} boolState - The boolean value to update the state with.
-   * @return {void} This function does not return a value.
-   */
-  const updateState = (index, boolState) => {
-    let states = state;
-    states = states.map((state, iterator) => {
-      return iterator === index ? boolState : state;
-    });
-    setState(states);
-  };
-
-  /**
-   * Updates the status of an item at the specified index.
-   *
-   * @param {number} index - The index of the item to update.
-   * @return {void} This function does not return a value.
-   */
-  const handleChangeStatus = (index) => {
-    updateState(index, true);
-    if (type === 'pack') {
-      dispatch(changePackStatus({ _id, is_public: !is_public }));
-    } else if (type === 'trip') {
-    }
-  };
+  const { state, setState, handleChangeStatus } = useUserDataCard(
+    type,
+    _id,
+    is_public,
+    index,
+    differentUser,
+  );
 
   const truncatedName = truncateString(name, 25);
   const truncatedDestination = truncateString(destination, 25);
 
   return (
-    <View style={{ alignItems: 'center', padding: '16px' }}>
+    <View style={{ alignItems: 'center', padding: 16 }}>
       <View
         style={{
-          minHeight: '150px',
-          minWidth: '300px',
-          border: '1px solid gray',
-          borderLeft: `10px solid ${is_public ? 'green' : 'red'}`,
-          borderRadius: '8px',
+          minHeight: 150,
+          minWidth: 300,
+          border: 1, 'solid gray',
```

Comment:
be careful this syntax changes the styling i believe when you do comma separated